src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtils.cs

```diff
@@ -102,6 +113,51 @@ internal static void SetHttpTagsFromResult(Activity? activity, object? result)
     internal static string? GetQueryString(APIGatewayHttpApiV2ProxyRequest request) =>
         string.IsNullOrEmpty(request.RawQueryString) ? string.Empty : "?" + request.RawQueryString;
 
+    internal static string? GetQueryString(ApplicationLoadBalancerRequest request)
+    {
+        // If the request has a query string value, one of the following properties will be set, depending on if
+        // "Multi value headers" is disabled on ELB Target Group.
+        if (request.QueryStringParameters != null)
```

Comment:
I think it needs to be the other way round, i.e. you should check `MultiValueQueryStringParameters` first, at least that would match other event types. But please check what happens if you e.g. trigger an ALB event with a query like `?a=v1&a=v2&b=vb`. I would expect that everything is also contained in MultiValueQueryStringParameters while QueryStringParameters might only have partial information for key `a`.

Reply:
As far as I'm aware, these two fields are mutually exclusive so we shouldn't see a situation where both are set for the same request. I agree though that it makes sense to prioritise MultiValue here.

https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#enable-multi-value-headers

Reply:
I've tested both with multi value headers enabled and disabled on the target group, and it behaves as you've described.